pyfr/plugins/fluidforce.py

```diff
@@ -172,15 +206,28 @@ def __call__(self, intg):
 
                 # Do the quadrature
                 f[ndims:] += np.einsum('i...,klij,jil', qwts, vis, norms)
+                if self._mcomp:
+                    # Get the relative flux points position
+                    rfpts = self._rfpts[etype, fidx]
+
+                    # Viscous force at each flux point
+                    viscf = np.einsum('ijkl,lkj->lki', vis, norms)
+
+                    # Viscous force moments
+                    mop = 'i,ji->' if self.ndims == 2 else 'i,jik->k'
+                    m[self._mcomp:] += np.einsum(mop, qwts,
```

Comment:
What does the non-einsum (broadcast-based) solution look like?

Reply:
You mean for the cross product or to multiply `qwts` by the cross product?

I've changed this part to avoid the if in `mop`.